src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,23 @@
 package practice;
 
-public class CandidateValidator {
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final String TARGET_NATIONALITY = "Ukrainian";
+    private static final int MIN_YEARS_PERIOD = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] years = candidate.getPeriodsInUkr().split("-");
```

Comment:
you can simlify the code
Create separate method `checkTimeLivingInCountry`
```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(REQUIRED_NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```

Reply:
please fix comment